File: P1/P1/Ejercicio3/PersonajeBuilder.py
from abc import ABC, abstractmethod
from Personaje import Personaje

class PersonajeBuilder(ABC):
    def __init__(self):
        self.personaje = None

    def create_new_personaje(self):
        self.personaje = Personaje()

    @abstractmethod
    def add_armadura(self):
        pass
 
    @abstractmethod
    def add_arma(self):
        pass

    @abstractmethod
    def add_habilidad(self):
        pass
    """ prueba """
    @abstractmethod
    def get_armadura(self):
        pass

File: P2-P3/ejercicio3/lib/modelo/personaje_builder.dart
import 'package:ejercicio3/modelo/personaje.dart';
import 'package:ejercicio3/modelo/armadura.dart';

abstract class PersonajeBuilder {
  // La interrogación es para que el atributo pueda ser null
  Personaje? personaje;
  
  String? usuario;

  PersonajeBuilder() {
    personaje = null;
  }

  void createNewPersonaje(String? tipoPersonaje) {
    personaje = Personaje();
    personaje!.tipoPersonaje = tipoPersonaje;
  }

  //MEJORAS
  void setArmadura(Armadura armadura);

  Armadura? getArmadura();

  ////////////////////////

  //Metodo a implementar
  void addArma();

  //Metodo a implementar
  void addArmadura();

  //Metodo a implementar
  void addHabilidad();
}
